infra/docker/server.Dockerfile

```
@@ -1,48 +1,56 @@
+
+# 3. Install all dependencies
 RUN pnpm install --frozen-lockfile
 
-# Copy source code
+# 4. Copy the rest of the source code
 COPY . .
 
-# Build shared types, generate prisma client, and build server (single layer)
+# 5. Build shared types and the server
 ARG DATABASE_URL="postgresql://placeholder:placeholder@localhost:5432/placeholder"
 RUN pnpm --filter @shared/types build && \
```

Hadolint (2.14.0)

[warning] Use WORKDIR to switch to a directory

(DL3003)